internals/controllers/rolebinding/rolebinding_cleanup.go

```diff
@@ -0,0 +1,108 @@
+package controllers
+
+import (
+	"fmt"
+	danav1 "github.com/dana-team/hns/api/v1"
+	"github.com/dana-team/hns/internals/utils"
+	"github.com/go-logr/logr"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// cleanUp takes care of clean-up related operations that need to be done when
+// a roleBinding relevant to HNS is deleted
+func (r *RoleBindingReconciler) cleanUp(rbObject *utils.ObjectContext, sns *utils.ObjectContextList) error {
+	ctx := rbObject.Ctx
+	logger := log.FromContext(ctx)
+	logger.Info("cleaning up roleBinding")
+
+	rbName := rbObject.Object.GetName()
+	rbNamespace := rbObject.Object.GetNamespace()
+
+	if !IsServiceAccount(rbObject.Object) {
```

Comment:
put this logic in function it will help explain why you did that

Reply:
Not sure I agree it'd be clearer?